### PR TITLE
Fix graceful shutdown skipped if future ends with error

### PR DESCRIPTION
**Pull request by cecton**
I just noticed this error while working on the task manager

### CODE REVIEW

File: client/cli/src/runner.rs [hunk 1]
```diff
@@ -229,10 +229,9 @@ impl<C: SubstrateCli> Runner<C> {
 	) -> Result<()> {
 		self.print_node_infos();
 		let mut task_manager = initialise(self.config)?;
-		self.tokio_runtime.block_on(main(task_manager.future().fuse()))
-			.map_err(|e| e.to_string())?;
+		let res = self.tokio_runtime.block_on(main(task_manager.future().fuse()));
 		self.tokio_runtime.block_on(task_manager.clean_shutdown());
```
Comment (tomaka) [inline client/cli/src/runner.rs:233]: This is a recurring question, but *why* do we even need a clean shut down? 
It feels to me that someone one day has randomly put a line that waits for the shutdown, and nobody has ever dared remove it.

Comment (cecton) [reply client/cli/src/runner.rs:233]: TL;DR If we want to support any async runtime, we need to handle tasks cancelling.

All the details are here: https://github.com/paritytech/substrate/issues/6642

There is also a link to the issue on async-std on the matter.

Comment (cecton) [reply client/cli/src/runner.rs:233]: If the question is more about why not just let the tasks be... If they are not cancelled, their objects are not properly dropped (Drop is not called).

**Comment by cecton** [comment]
bot merge

**Comment by cecton** [comment]
bot merge

File: client/cli/src/runner.rs [hunk 2]
```diff
@@ -229,10 +229,9 @@ impl<C: SubstrateCli> Runner<C> {
...
-		Ok(())
+		Ok(res.map_err(|e| e.to_string())?)
```
Comment (bkchr) [inline client/cli/src/runner.rs:234]:
```suggestion
		res.map_err(|e| e.to_string())
```

Comment (cecton) [reply client/cli/src/runner.rs:234]: That doesn't work. I will a version with .into()

File: client/cli/src/runner.rs (hunk 1 above)
Comment (andresilva) [reply client/cli/src/runner.rs:233]: The main issue (that came from parity-ethereum) was rocksdb corruption on improper shutdown. Don't remember if it has happened in substrate.

**Comment by cecton** [comment]
bot merge

File: client/cli/src/runner.rs (hunk 1 above)
Comment (cecton) [reply client/cli/src/runner.rs:233]: That rings a bell. Maybe @arkpar would have more info on that?

**Pull request merged**